### ISSUE TITLE
Training loss 0 while finetuning

### ISSUE THREAD

**Issue by khushboo-anand0909** [haotian-liu/LLaVA#375]
### Question

Hello, I'm trying to finetune the model on top of your provided checkpoints on my custom instruction dataset. However, I'm getting train loss 0 throughout all epochs while lora finetuning using finetune_lora.sh What can be the possible reason?

**Comment by kanshichao** [comment]
I conducted pretraining of the model utilizing the 'zeros2.json' dataset, followed by subsequent model finetuning employing the 'zero3_offload.json' dataset, where a loss of 0 was similarly achieved during the finetuning phase. The batch_size is set to 32*4 on both stages.

**Comment by Fake10086** [comment]
hello, have you solved the problem?

**Comment by Fake10086** [comment]
Hi, I've solved the problem. If this 'loss equals zero' problem relates with some tokenization mismatch which you get some warning, I believe it is the same problem that you are NOT choosing the corresponding version of Model and Tokenizer.

**Comment by haotian-liu** [comment]
It seems that most issues of loss=0 is related to the tokenizer version mismatch. Please try installing with the corresponding versions. Thanks.

**Comment by PCIResearch** [comment]
> Hi, I've solved the problem. If this 'loss equals zero' problem relates with some tokenization mismatch which you get some warning, I believe it is the same problem that you are NOT choosing the corresponding version of Model and Tokenizer.

I would like to ask you how you solved this problem?

**Comment by wanghao9610** [comment]
Somebody who encountered the issue may ref this issue (https://github.com/haotian-liu/LLaVA/issues/1231) for help.

**Comment by Logos23333** [comment]
After setting the args `model_max_length` to 512, the loss becomes 0. When set to 2048, the loss is normal, so there may be an issue with the tokenizer.

**Comment by eslambakr** [comment]
Thanks, @Logos23333. This was my issue as I was using another Image encoder that exceeded the model_max_length.

**Comment by Ayush-00** [comment]
> Somebody who encountered the issue may ref this issue (#1231) for help.

Thanks @wanghao9610 for the reference. My `model_max_length` was already 2048, but this fixed my issue!

**Comment by OmarWill1** [comment]
same probleme ( mt5-small ) decoder encoder architecture
